pkg/host/internal/network/network.go

```diff
@@ -95,10 +96,20 @@ func (n *network) TryGetInterfaceName(pciAddr string) string {
 		return name
 	}
 
-	log.Log.V(2).Info("tryGetInterfaceName()", "name", netDevName)
+	log.Log.V(2).Info("TryGetInterfaceName()", "name", netDevName)
 	return netDevName
 }
 
+// TryGetInterfaceName tries to find the SR-IOV virtual interface index base on pci address
+func (n *network) TryGetInterfaceIndex(pciAddr string) int {
+	ifIndex, err := n.dputilsLib.GetNetIndex(pciAddr)
```

Comment:
you may get the wrong index in case of switched since implementation returns the first netdev interface index found.

in TryGetInterfaceNames , if in switchdev we return the "uplink representor" netdev.  since in switchdev all representors will be found under the PF /net folder.

Reply:
in device plugin we also have GetPFNames()[1] which also takes switchdev into account

[1] https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/blob/fe9f2f85553267c884dc1e68b94e8561c57e8c50/pkg/utils/utils.go#L72